ports.js

```diff
@@ -1,51 +1,228 @@
 /* eslint-disable no-console */
 /* globals encrypt:false, decrypt:false */
+const CONTENT_KEY = "pad";
 
-let app;
-if (typeof Elm === 'undefined') {
-  // This happens if we're in the context of Electron.
-  const Elm = require('./hoverpad.js');
-  app = Elm.Main.fullscreen();
-} else {
-  app = Elm.Main.fullscreen();
+const IS_WEB_EXTENSION = (typeof chrome === "object" && typeof chrome.storage === "object");
+
+function storePassphrase(passphrase) {
+  // Insecurely storing the passphrase.
+  sessionStorage.setItem("temporaryPassphrase", btoa(passphrase));
+  return Promise.resolve(null);
+}
+
+function dropPassphrase() {
+  sessionStorage.removeItem("temporaryPassphrase");
+  return Promise.resolve(null);
 }
 
-app.ports.decryptData.subscribe(function(data) {
-  console.log(data);
-  if (!data.content) {
-    app.ports.dataDecrypted.send(null);
-    return;
+function getPassphrase() {
+  // Insecurely retrieving the passphrase.
```

Comment:
nit: *Obfuscating the stored passphrase to avoid exposing it in plain text to potential readers, but this should not be considered as secure.*